File: src/main/java/social/alone/server/common/controller/BaseController.java
package social.alone.server.common.controller;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.validation.Errors;
import social.alone.server.common.validate.ErrorsResource;

public class BaseController {

  protected ResponseEntity BadRequest(Errors errors) {
    return ResponseEntity.badRequest().body(new ErrorsResource(errors));
  }

  protected ResponseEntity NotFound() {
    return ResponseEntity.notFound().build();
  }

  protected ResponseEntity forbidden(){
    return ResponseEntity.status(HttpStatus.FORBIDDEN).build();
  }
}

File: src/main/kotlin/social/alone/server/controller/BaseController.kt
package social.alone.server.controller

import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.validation.Errors
import social.alone.server.validate.ErrorsResource

open class BaseController {

    protected fun BadRequest(errors: Errors): ResponseEntity<*> {
        return ResponseEntity.badRequest().body(ErrorsResource(errors))
    }

    protected fun NotFound(): ResponseEntity<*> {
        return ResponseEntity.notFound().build<Any>()
    }

    protected fun forbidden(): ResponseEntity<*> {
        return ResponseEntity.status(HttpStatus.FORBIDDEN).build<Any>()
    }
}
